Lib/idlelib/idle_test/mock_idle.py

```diff
@@ -40,8 +40,9 @@ def __call__(self, *args, **kwds):
 class Editor:
     '''Minimally imitate editor.EditorWindow class.
     '''
-    def __init__(self, flist=None, filename=None, key=None, root=None):
-        self.text = Text()
+    def __init__(self, flist=None, filename=None, key=None, root=None,
+                 text=None):  # Allow real Text with mock Editor.
```

Comment:
IMO this change in unnecessary, since one may simply override the `.text` attribute after creating an `Editor` instance.

Reply:
True.  Either way works.  Today I slightly prefer using dependency injection to not create and delete an unneeded mock Text.  I am more concerned about the noise of the unused parameters and the missing attributes that resulted in a second mock Editor being defined for some of the format tests.  But I want to leave that for another issue.